pmd-xml/src/main/java/net/sourceforge/pmd/lang/xml/ast/XmlParser.java

```diff
@@ -65,35 +64,31 @@ protected Document parseDocument(Reader reader) throws ParseException {
         }
     }
 
+
     public XmlNode parse(Reader reader) {
         Document document = parseDocument(reader);
-        return createProxy(document);
+        XmlNode root = new RootXmlNode(this, document);
+        nodeCache.put(document, root);
+        return root;
     }
 
-    public XmlNode createProxy(Node node) {
-        XmlNode proxy = nodeCache.get(node);
-        if (proxy != null) {
-            return proxy;
-        }
 
-        // TODO Change Parser interface to take ClassLoader?
-        LinkedHashSet<Class<?>> interfaces = new LinkedHashSet<>();
-        interfaces.add(XmlNode.class);
-        if (node instanceof Document) {
-            interfaces.add(RootNode.class);
+    public XmlNode wrapDomNode(org.w3c.dom.Node domNode) {
```

Comment:
why is this method public?

Reply:
I didn't think about it, it could be package private